src/main/java/com/tananaev/jsonpatch/operation/ReplaceOperation.java

```diff
@@ -22,10 +22,9 @@ public String getOperationName() {
     }
 
     @Override
-    public JsonElement apply(JsonElement original) {
-        JsonElement result = duplicate( original );
+    public void applyInPlace(InPlaceElementWrapper inPlaceElement) {
```

Comment:
Can't we just have a return value in the `applyInPlace` instead of having a wrapper?

Reply:
If we return a value, we won't actually modifying the input inPlace in all the cases. So, we always have to make sure that we read the return value and not the passed input. (when we need conversion across JsonElement types we will have to return a different object as we cannot actually modify the passed input)

It will be a little ambiguous here as people would generally expect any function that modifies input InPlace to have the same value either by reading the return value or reading the passed input.

Doing it with a wrapper here ensures that there is a single way to read the final result and read the correct result.